src/components/dialogs/contingency-list/contingency-list-utils.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import {
+    CONTINGENCY_LIST_TYPE,
+    CONTINGENCY_NAME,
+    COUNTRIES_1,
+    COUNTRIES_2,
+    CRITERIA_BASED,
+    EQUIPMENT_IDS,
+    EQUIPMENT_TABLE,
+    EQUIPMENT_TYPE,
+    NAME,
+    NOMINAL_VOLTAGE_1,
+    NOMINAL_VOLTAGE_2,
+    AG_GRID_ROW_UUID,
+    SCRIPT,
+} from '../../utils/field-constants';
+import { ContingencyListType } from '../../../utils/elementType';
+import { prepareContingencyListForBackend } from '../contingency-list-helper';
+import {
+    saveCriteriaBasedContingencyList,
+    saveExplicitNamingContingencyList,
+    saveScriptContingencyList,
+} from '../../../utils/rest-api';
+import { DEFAULT_RANGE_VALUE } from '../../utils/rhf-inputs/range-input';
+import {
+    getCriteriaBasedFormData,
+    getCriteriaBasedSchema,
+} from '../commons/criteria-based/criteria-based-utils';
+import yup from '../../utils/yup-config';
+import { getExplicitNamingSchema } from './explicit-naming/explicit-naming-form';
+
+export const DEFAULT_ROW_VALUE = {
+    [CONTINGENCY_NAME]: '',
+    [EQUIPMENT_IDS]: [],
+};
+export const DEFAULT_TABLE_ROWS = [
+    DEFAULT_ROW_VALUE,
+    DEFAULT_ROW_VALUE,
+    DEFAULT_ROW_VALUE,
+];
+
+export const getContingencyListSchema = () =>
+    yup.object().shape({
+        [NAME]: yup.string().required(),
+        [CONTINGENCY_LIST_TYPE]: yup.string().nullable(),
+        [SCRIPT]: yup.string().nullable(),
+        ...getExplicitNamingSchema(EQUIPMENT_TABLE),
+        ...getCriteriaBasedSchema(),
+    });
+
+export const getContingencyListEmptyFormData = () => ({
+    [NAME]: '',
+    [EQUIPMENT_TABLE]: DEFAULT_TABLE_ROWS,
+    [CONTINGENCY_LIST_TYPE]: ContingencyListType.CRITERIA_BASED.id,
+    [SCRIPT]: '',
+    ...getCriteriaBasedFormData(),
+});
+
+export const getFormDataFromFetchedElement = (
+    response,
+    name,
+    contingencyListType
+) => {
+    switch (contingencyListType) {
+        case ContingencyListType.CRITERIA_BASED.id:
+            return {
+                [NAME]: name,
+                ...getCriteriaBasedFormData(
+                    CRITERIA_BASED,
+                    response.equipmentType,
+                    response.countries1,
+                    response.countries2,
+                    response?.nominalVoltage1 ?? DEFAULT_RANGE_VALUE,
```

Comment:
Response is supposed to be null sometimes ? I don't think so

Reply:
fixed

Reply:
There is still some `response?`

Reply:
all removed